contracts/automation/andromeda-evaluation/src/contract.rs

```diff
@@ -0,0 +1,1045 @@
+use crate::state::{CONDITION_ADO_ADDRESS, QUERY_ADO_ADDRESS, TASK_BALANCER_ADDRESS};
+use ado_base::state::ADOContract;
+use andromeda_automation::evaluation::{
+    ExecuteMsg, InstantiateMsg, MigrateMsg, Operators, QueryMsg,
+};
+use andromeda_automation::task_balancer::ExecuteMsg::RemoveProcess;
+use common::{
+    ado_base::InstantiateMsg as BaseInstantiateMsg, app::AndrAddress, encode_binary,
+    error::ContractError,
+};
+use cosmwasm_std::{
+    ensure, entry_point, to_binary, Binary, CosmosMsg, Deps, DepsMut, Env, MessageInfo,
+    QueryRequest, Reply, Response, StdError, SubMsg, Uint128, WasmMsg, WasmQuery,
+};
+use cw2::{get_contract_version, set_contract_version};
+use cw_utils::nonpayable;
+use semver::Version;
+
+// version info for migration info
+const CONTRACT_NAME: &str = "crates.io:andromeda-evaluation";
+const CONTRACT_VERSION: &str = env!("CARGO_PKG_VERSION");
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn instantiate(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    msg: InstantiateMsg,
+) -> Result<Response, ContractError> {
+    set_contract_version(deps.storage, CONTRACT_NAME, CONTRACT_VERSION)?;
+
+    CONDITION_ADO_ADDRESS.save(deps.storage, &msg.condition_address)?;
```

Comment:
Any contracts that use `AndrAddress`es should validate that they are valid within the app, see [here](https://github.com/andromedaprotocol/andromeda-core/blob/development/contracts/finance/andromeda-splitter/src/contract.rs#L97)

Reply:
This is also true for some other contracts  within this PR